src/app/self-service/users/user.service.ts

```diff
@@ -54,4 +55,36 @@ export class UserService {
     return this.http.disableApiPrefix().get('/assets/mock/user.mock.json');
   }
 
+  /**
+   * Change User Password
+   * @param userId User Id of users
+   * @param password New Password of the user
+   * @returns {Observable<any>}
+   */
+  changePassword(userId: string, passwordObj: any) {
+    return this.http.put(`/users/${userId}`, passwordObj);
+  }
+
+  /**
+   * Confirm Change Password of Users
+   * @param controlNameToCompare
+   */
+  confirmPassword(controlNameToCompare: string): ValidatorFn {
+    return (c: AbstractControl): ValidationErrors|null => {
+        if (c.value == null || c.value.length === 0) {
+            return null;
+        }
+        const controlToCompare = c.root.get(controlNameToCompare);
+        if (controlToCompare) {
+            const subscription: Subscription = controlToCompare.valueChanges.subscribe(() => {
+                c.updateValueAndValidity();
+                subscription.unsubscribe();
+            });
+        }
+        return controlToCompare && controlToCompare.value !== c.value ? {'notequal': true} : null;
+    };
+  }
+
```

Comment:
I think we should use a password service for both these functions

Reply:
Yes, In #660 I put the function in component.ts . Do we have to create a new service file and put it for the reusability?

Reply:
yes